syntax = "proto3";

package redisgrpc;


service RedisGrpcServer{
  rpc InitConnection(InitRequest) returns (InitReply){};
  rpc Set(SetRequest) returns (SetReply) {};
  rpc Get(GetRequest) returns (GetReply) {};
  rpc Shutdown(ShutdownRequest) returns (ShutdownReply){};
  rpc GetCacheForLogging(LoggerGetRequest) returns (LoggerGetReply){};

}


message InitRequest {
   string connection_id = 1;
}


message InitReply {
   string status = 1;
}


message ShutdownRequest {

}


message ShutdownReply {
  string status = 1;
}


message SetRequest{
  string key = 1;
  string value = 2;
}

message SetReply {
  string status = 1;
}


message GetRequest{
  string key = 1;
}


message GetReply {
  string status = 1;
  string value = 2;
}

message LoggerGetRequest{

}

message LoggerGetReply{
  string status = 1;
  map<string, string> cache = 2;
}
